### PR TITLE
Issue #2651 - Updates case symptoms when creating/editing a visit

### LINKED ISSUE #2651: [FRENCH TEAM] Updating the case symptoms based on visit symptoms (case follow-up) [0.5]
### Situation Description
#1886 has added follow-up visits to cases.

### Feature Description
When a new visit is created for a case, the symptoms of the case should automatically be updated.
See clinical visits for comparison.

### PR DESCRIPTION
**Pull request by xavier-calland**
Fixes #2651

### CODE REVIEW

File: sormas-backend/src/main/java/de/symeda/sormas/backend/visit/VisitFacadeEjb.java [hunk 1]
```diff
@@ -623,7 +627,11 @@ private void onVisitChanged(VisitDto existingVisit, Visit newVisit) {
 		}
 
 		if (newVisit.getCaze() != null) {
-			caseService.updateFollowUpUntilAndStatus(newVisit.getCaze());
```
Comment (MateStrysewske) [inline sormas-backend/src/main/java/de/symeda/sormas/backend/visit/VisitFacadeEjb.java:626]: What's the reason for removing this line of code?

File: sormas-backend/src/main/java/de/symeda/sormas/backend/visit/VisitFacadeEjb.java [hunk 2]
```diff
@@ -623,7 +627,11 @@ private void onVisitChanged(VisitDto existingVisit, Visit newVisit) {
...
+			// Update case symptoms
+			CaseDataDto caze = CaseFacadeEjb.toDto(newVisit.getCaze());
+			SymptomsDto caseSymptoms = caze.getSymptoms();
+			SymptomsHelper.updateSymptoms(toDto(newVisit).getSymptoms(), caseSymptoms);
+			caseFacade.saveCase(caze);
```
Comment (MateStrysewske) [inline sormas-backend/src/main/java/de/symeda/sormas/backend/visit/VisitFacadeEjb.java:634]: Please also call SymptomsHelper.updateIsSymptomatic before saving the case

File: sormas-backend/src/test/java/de/symeda/sormas/backend/caze/CaseFacadeEjbTest.java [hunk 3]
```diff
@@ -1022,6 +1024,40 @@ public void testDoesEpidNumberExist() {
...
+		// Update an existing visit
+		visit.getSymptoms().setAcuteRespiratoryDistressSyndrome(SymptomState.YES);
+		getVisitFacade().saveVisit(visit);
+
+		updatedCase = getCaseFacade().getCaseDataByUuid(caze.getUuid());
+
+		assertEquals(SymptomState.YES, updatedCase.getSymptoms().getChestPain());
+		assertEquals(SymptomState.YES, updatedCase.getSymptoms().getAbdominalPain());
+		assertEquals(SymptomState.YES, updatedCase.getSymptoms().getAcuteRespiratoryDistressSyndrome());
+
```
Comment (MateStrysewske) [inline sormas-backend/src/test/java/de/symeda/sormas/backend/caze/CaseFacadeEjbTest.java:1058]: Add an assertion here that tests whether "symptomatic" is true

**Review by MateStrysewske — CHANGES_REQUESTED** [review]

File: sormas-backend/src/main/java/de/symeda/sormas/backend/visit/VisitFacadeEjb.java (hunk 2 above)
Comment (xavier-calland) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/visit/VisitFacadeEjb.java:634]: OK

File: sormas-backend/src/main/java/de/symeda/sormas/backend/visit/VisitFacadeEjb.java (hunk 1 above)
Comment (xavier-calland) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/visit/VisitFacadeEjb.java:626]: `caseFacade.saveCase()` will call `updateFollowUpUntilAndStatus`

File: sormas-backend/src/test/java/de/symeda/sormas/backend/caze/CaseFacadeEjbTest.java (hunk 3 above)
Comment (xavier-calland) [reply sormas-backend/src/test/java/de/symeda/sormas/backend/caze/CaseFacadeEjbTest.java:1058]: OK

File: sormas-backend/src/main/java/de/symeda/sormas/backend/visit/VisitFacadeEjb.java (hunk 1 above)
Comment (MateStrysewske) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/visit/VisitFacadeEjb.java:626]: Makes sense!

File: sormas-backend/src/main/java/de/symeda/sormas/backend/visit/VisitFacadeEjb.java (hunk 2 above)
Comment (xavier-calland) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/visit/VisitFacadeEjb.java:634]: https://github.com/hzi-braunschweig/SORMAS-Project/blob/482e8a9d7413a5ca591ffafdde4f29544077816c/sormas-backend/src/main/java/de/symeda/sormas/backend/caze/CaseFacadeEjb.java#L1267-L1272

I realise `SymptomsHelper.updateIsSymptomatic` is called in `saveCase` too.

Comment (MateStrysewske) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/visit/VisitFacadeEjb.java:634]: Alright, makes sense.

**Review by MateStrysewske — APPROVED** [review]

**Pull request merged**